libvirt/domain.go

```diff
@@ -280,17 +280,103 @@ func inlineUpdateCoreOsIgnition(d *schema.ResourceData, domainDef *libvirtxml.Do
 		if err != nil {
 			return err
 		}
-		err = guestfishExecution(rootPath, fileToUpload)
+		connURI, err := virConn.GetURI()
+		if err != nil {
+			return err
+		}
+		err = guestfishExecution(rootPath, fileToUpload, connURI)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func guestfishExecution(rootPath string, fileToUpload string, connURI string) error {
+	// get remote host from libvirtd remote URI
+	virHost := strings.Split(strings.SplitAfter(connURI, "//")[1], "/")[0]
```

Comment:
Slightly sceptical of this host extraction code but in my limited testing, it works fine: https://play.golang.org/p/Sj1UYFc2LSf . Only thing is that you get the [authority component of the URI](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Generic_syntax), which could be more than the hostname (in most cases a `username@host`) so please ensure the code doesn't break on that case.

Reply:
Oh and it doesn't account for local URLs with hostname (e.g qemu://localhost/system). You'd want a regex match like we did in Boxes: https://gitlab.gnome.org/GNOME/gnome-boxes/blob/master/src/machine.vala#L66

Reply:
Agreed, I'll add a helper function to get libvirt host, by considering more cases.

Reply:
Fixed this.